breadcrumbs-ui.js

```diff
@@ -0,0 +1,51 @@
+define([
+    '$'
+], function($) {
+
+    var init = function() {
+        var $breadcrumbs = $('.c-breadcrumbs');
+
+        $breadcrumbs.each(function() {
+            var $this = $(this);
+            var $links = $this.find('.c-breadcrumbs__item');
+            var showMoreText = "Show more breadcrumbs items";
+            var $showMore = $('<li class="c-breadcrumbs__item"><button class="c-breadcrumbs__link c--show-more js-breadcrumb-show" type="button">' + showMoreText + '</button></li>');
+            var containerWidth = $this.width();
+            var linkWidth = 0;
+
+            // Calculating the total width of the items
+            $links.each(function() {
+                linkWidth = linkWidth + $(this).width();
```

Comment:
As noted elsewhere, this might not be accurate if there is margin between items. There are a couple of other ways to calculate this. I’m not entirely sure which is best, but this might be the one I’d recommend:

```javascript
// $el is the element passed in by the Adaptive init function:
var el = $el.get(0); // unwrap jQuery from the element, getting the raw DOM node
var contentWidth = el.offsetLeft - el.parentNode.offsetLeft + $el.width();
```

This gets element’s distance from the left edge of its parent, then adds its own width to get the rightmost extent of the items. This should correspond to the width of all items, even with margin between.